### ISSUE TITLE
[flutter-in-app-purchase]  PlatformException (PlatformException(storekit_no_receipt, Cannot find receipt for the current main bundle., null, null))

### ISSUE THREAD

**Issue by wiktoriasobczyk** [flutter/flutter#74413]
I try to implement in-app purchases in my app and I've received this kind of error on iPhone 12 (iOS14). This problem doesn't occur in iPhone 11. Do you have any ideas to solve this problem?

`PlatformException (PlatformException(storekit_no_receipt, Cannot find receipt for the current main bundle., null, null))`

**Comment by TahaTesser** [comment]
Hi @wiktoriasobczyk 
Can you please provide your `flutter doctor -v` your `pubspec.yaml` and a complete reproducible minimal code sample
Thanks for filing the issue

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by tuanbs** [comment]
I runt the example of this plugin and I got the same error when it hits the following line:
```
final QueryPurchaseDetailsResponse purchaseResponse = await _connection.queryPastPurchases();
```
I use `in_app_purchase` version `0.5.1`. Please suggest solution or workaround. Thanks in advance.

**Comment by BringTheChill** [comment]
I am in the same situation @tuanbs and @wiktoriasobczyk are. Please help us, we really need it, because we ca not move our app to production if it's not fixed.
This is my flutter doctor -v:
`[✓] Flutter (Channel stable, 2.0.3, on macOS 11.2.3 20D91 darwin-x64, locale en-GB)
    • Flutter version 2.0.3 at /Users/robert/Developer/flutter
    • Framework revision 4d7946a68d (11 days ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/robert/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (3 available)
    • iPhone (mobile)    • 7f230f8009a52e248fb4b425069f9975c11ed11c • ios            • iOS
      14.4.1
    • iPhone 12 (mobile) • FCB20924-A8F8-4AF1-A2AF-3105A8DF4031     • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)       • chrome                                   • web-javascript •
      Google Chrome 89.0.4389.90

• No issues found!
`

**Comment by mg7237** [comment]
Same with me

**Comment by MaxAltena** [comment]
Happening to me as well, I have no idea how to fix this. This literally breaks my app.

**Comment by slobodanmilinkovic** [comment]
Same here. Breaks my apps. Please help.

**Comment by amandaoneal** [comment]
I was running into this too, and I think I figured it out. This seems to happen when there is no logged-in Apple account.

To fix:

1. Delete the app from the test device entirely
2. Launch the app **from XCode**, not from your Flutter IDE
3. It should pop up asking for your Apple username/password
4. Enter a valid username/password
5. Now you should be able to launch from your Flutter IDE without this crash

**Comment by ride4sun** [comment]
@amandaoneal  this did not work for me> I still get the same exception. 
Amy help from the plugin guys?

**Comment by TylerSustare** [comment]
I'll pile on. This is also on my physical phone that I can't follow @amandaoneal advice because it doesn't ask for my apple username/pass again

**Comment by ride4sun** [comment]
Yes me too - It's a physical Phone. Maybe that makes a difference.... I can not run it in the simulator because I need the Bluetooth connection

**Comment by mg7237** [comment]
After multiple rounds of testing, I couldnt get a way around this issue. I migrated to https://pub.dev/packages/flutter_inapp_purchase .  Took me not more than 4 hours considering the configurations were already done to support in app purchase.

**Comment by jdragonball** [comment]
Same here....

**Comment by kdcyberdude** [comment]
Any update on this?
